consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +498,27 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// If no cached result, we import block through fetcher, which requires processing the block one more time.
+	// Else if cache hit, we directly write the block.
+	if result == nil {
+		sb.logger.Warn("state process cache miss, enqueuing the block to fetcher", "number", block.Number(), "hash", block.Hash())
+		if sb.broadcaster != nil {
+			sb.broadcaster.Enqueue(fetcherID, block)
+		}
+		sb.logger.Warn("Cannot import consensus block", "number", block.Number(), "hash", block.Hash())
+		return nil
+	}
+
+	if block := sb.chain.GetBlock(block.Hash(), block.NumberU64()); block != nil {
```

Comment:
Is there really an issue with calling `writeBlockWithState` for an already persisted block? If that's the case we have a bigger problem.

This check here, won't save us, since there's a race condition on it. And the proposer already has the same race condition.